### PR TITLE
Register all methods of Liquibase services sub-classes for reflection

### PR DESCRIPTION
**Pull request by andrejpetras**
Issue: #8357

### CODE REVIEW

File: extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java [hunk 1]
```diff
@@ -187,11 +188,11 @@ ServiceStartBuildItem configureRuntimeProperties(LiquibaseRecorder recorder,
     private void addService(BuildProducer<ReflectiveClassBuildItem> reflective,
             BuildProducer<GeneratedResourceBuildItem> generatedResourceProducer,
             BuildProducer<NativeImageResourceBuildItem> resourceProducer,
-            Class<?> className) {
+            Class<?> className, boolean methods) {
 
         Class<?>[] impl = ServiceLocator.getInstance().findClasses(className);
         if (impl != null && impl.length > 0) {
-            reflective.produce(new ReflectiveClassBuildItem(true, false, false, impl));
+            reflective.produce(new ReflectiveClassBuildItem(true, methods, false, impl));
```
Comment (gsmet) [inline extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java:195]: I think I would make it true for all services.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java (hunk 1 above)
Comment (andrejpetras) [reply extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java:195]: It was my first solution, but I'm getting errors for some classes at native compile time.
I'll try again and create two class lists.

Does it make big different for the native image (compile time) if we register all the classes?

Comment (andrejpetras) [reply extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java:195]: Done, only the `LicenseService` class needs to be `methods=false`

Comment (gsmet) [reply extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java:195]: So doing that might include more methods than strictly needed in the native image but I think it's safer.

As for the `LicenseService` it's probably related to the substitutions you had to make there so looks OK to me.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Nice work, thanks!

**Pull request merged**